src/ios/SpotifyOAuthPlugin.swift

```diff
@@ -13,15 +13,19 @@ extension URL {
     private var currentNsObserver: AnyObject?
     
     @objc(getCode:) func getCode(_ command: CDVInvokedUrlCommand) {
-        let auth = SPTAuth.defaultInstance()
+        let clientid = command.argument(at: 0) as! String
+        let redirectURL = URL(string: command.argument(at: 1) as! String)!
+        let tokenRefreshURL = URL(string: command.argument(at: 3) as! String)!
+        let requestedScopes = command.argument(at: 4) as! [String]
+        let redirectEncoded = redirectURL.absoluteString.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)!
         
-        auth.clientID = command.argument(at: 0) as! String
-        auth.redirectURL = URL(string: command.argument(at: 1) as! String)
-        auth.tokenSwapURL = URL(string: command.argument(at: 2) as! String)
-        auth.tokenRefreshURL = URL(string: command.argument(at: 3) as! String)
-        auth.requestedScopes = command.argument(at: 4) as! Array
+        var webUrl = "https://accounts.spotify.com/authorize?client_id="+clientid+"&response_type=code&redirect_uri="
+        webUrl += redirectEncoded
+        webUrl += "&show_dialog=true&scope="
+        webUrl += scopesToString(scopes: requestedScopes)
+        webUrl += "&utm_source=spotify-sdk&utm_medium=ios-sdk&utm_campaign=ios-sdk"
```

Comment:
I think there is a more idiomatic and fault-tolerant way to build URLs. Please see https://www.swiftbysundell.com/articles/constructing-urls-in-swift